RecoCTPPS/TotemRPLocal/src/CTPPSDiamondTrackRecognition.cc

```diff
@@ -0,0 +1,146 @@
+/****************************************************************************
+ *
+ * This is a part of CTPPS offline software.
+ * Authors:
+ *   Laurent Forthomme (laurent.forthomme@cern.ch)
+ *   Nicola Minafra (nicola.minafra@cern.ch)
+ *
+ ****************************************************************************/
+
+#include "RecoCTPPS/TotemRPLocal/interface/CTPPSDiamondTrackRecognition.h"
+
+#include <cmath>
+#include <cstdio>
+#include <algorithm>
+
+#define PAD_DEFAULT_FUNCTION "(x>[0]-0.5*[1])*(x<[0]+0.5*[1])"
+
+//----------------------------------------------------------------------------------------------------
+
+CTPPSDiamondTrackRecognition::CTPPSDiamondTrackRecognition() :
+  threshold_( 1.5 ), thresholdFromMaximum_( 0.5 ),
+  resolution_( 0.025 ), sigma_( 0.0 ),
+  startFromX_( -2 ), stopAtX_( 38 ),
```

Comment:
is there a reason these defaults are needed?
Can the constructor with PSet argument be the only valid option?